ts-rs/tests/enum_flattening.rs

```diff
@@ -0,0 +1,104 @@
+use serde::Serialize;
```

Comment:
We're getting some unused import warnings here.  
Also, we're not testing `#[ts(flatten)]`, which should be equivalent do `#[serde(flatten)]` with compat-mode on. 
I think with a few `#[cfg(..)]`s and `#[cfg_attr(..)]`s, we can make the test work with both compat-mode on and off as-is.

Reply:
Yeah, I'm adding those now, but the only test that will work without `serde-compat` is `internally_tagged` as all the others require `#[serde(tag = "...")]` or `#[serde(untagged)]`